### PR TITLE
Show progress bar when throwing items

### LINKED ISSUE #612: Show progress bar when throwing items
building upon #54, would like to see commands like /cenchant to show the progress bar in chat when throwing items to reach a certain seed. From my experience, some could throw items for more than 5 minutes. Would be nice if there's a progress bar in chat to show us the progress

And to further enhance that, it'd be great if a notification bell sound could be played (perhaps toggleable in config) when all items have been thrown and the seed has been reached. This would allow the player to be more AFK.

### PR DESCRIPTION
**Pull request by RealRTTV**
Closes #612

### CODE REVIEW

**Comment by RealRTTV** [comment]
lgtm

File: src/main/java/net/earthcomputer/clientcommands/features/EnchantmentCracker.java [hunk 1]
```diff
@@ -450,6 +454,26 @@ public boolean condition() {
...
+                                public void onCompleted() {
+                                    Minecraft.getInstance().player.playNotifySound(SoundEvents.NOTE_BLOCK_PLING.value(), SoundSource.PLAYERS, 1.0f, 2.0f);
+                                }
+
+                                @Override
+                                protected void onItemThrown(int current, int total) {
+                                    MutableComponent builder = Component.empty();
+                                    int color = Mth.hsvToRgb(current / (total * 3.0f), 1.0f, 1.0f);
+                                    builder.append(Component.literal("[").withColor(ChatFormatting.GRAY.getColor()));
+                                    builder.append(Component.literal("~" + Math.round(100.0 * current / total) + "%").withColor(color));
```
Comment (Earthcomputer) [inline src/main/java/net/earthcomputer/clientcommands/features/EnchantmentCracker.java:468]: You might want to cast the `Math.round` back to an `int` just so you don't get the `.0` in the string.

File: src/main/java/net/earthcomputer/clientcommands/features/EnchantmentCracker.java [hunk 2]
```diff
@@ -450,6 +454,26 @@ public boolean condition() {
...
+                                    builder.append(Component.literal("] ").withColor(ChatFormatting.GRAY.getColor()));
+                                    int filled_width = (int) Math.round((double) PROGRESS_BAR_WIDTH * current / total);
+                                    int unfilled_width = PROGRESS_BAR_WIDTH - filled_width;
```
Comment (Earthcomputer) [inline src/main/java/net/earthcomputer/clientcommands/features/EnchantmentCracker.java]: Please use camel case for variables

**Review by Earthcomputer — CHANGES_REQUESTED** [review]

File: src/main/java/net/earthcomputer/clientcommands/features/EnchantmentCracker.java (hunk 2 above)
Comment (RealRTTV) [reply src/main/java/net/earthcomputer/clientcommands/features/EnchantmentCracker.java]: again, sorry, idk why i did this

**Review by Earthcomputer — APPROVED** [review]

**Pull request merged**